src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp

```diff
@@ -0,0 +1,180 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace dg {
+/*!
+ * \brief Compute the Jacobian determinant times the inverse Jacobian so that
+ * the result is divergence-free.
+ *
+ * The metric identities are given by
+ *
+ * \f{align*}{
+ *
+ * \partial_{\hat{\imath}}\left(J\frac{\partial\xi^{\hat{\imath}}}
+ * {\partial x^i}\right)=0.
+ *
+ * \f}
+ *
+ * We want to compute \f$J\partial\xi^{\hat{\imath}}/\partial x^i\f$ in such a
+ * way that the above metric identity is satisfied numerically/discretely. We
+ * refer to the inverse Jacobian computed this way as the "metric
+ * indentity-satisfying inverse Jacobian".
+ *
+ * The general form is given by:
+ *
+ * \f{align*}{
+ *
+ * 2\left(J\frac{\partial \xi^{\hat{\imath}}}{\partial x^i}\right)_{\breve{s}}
+ *  &=\epsilon_{ijk}
+ *    \sum_{t_{\hat{1}}} \epsilon^{\hat{\imath}\hat{1}\hat{k}}
+ *    D^{(\hat{1})}_{\breve{s}\breve{t}}
+ *    \left(x^j\frac{\partial x^k}{\partial \xi^{\hat{k}}}\right)_{\breve{t}}
+ *    \notag \\
+ *  &+\epsilon_{ijk}
+ *    \sum_{t_{\hat{2}}} \epsilon^{\hat{\imath}\hat{2}\hat{k}}
+ *    D^{(\hat{2})}_{\breve{s}\breve{t}}
+ *    \left(x^j\frac{\partial x^k}{\partial \xi^{\hat{k}}}\right)_{\breve{t}}
+ *    \notag \\
+ *  &+\epsilon_{ijk}
+ *    \sum_{t_{\hat{3}}}\epsilon^{\hat{\imath}\hat{3}\hat{k}}
+ *    D^{(\hat{3})}_{\breve{s}\breve{t}}
+ *    \left(x^j\frac{\partial x^k}{\partial \xi^{\hat{k}}}\right)_{\breve{t}}.
+ *
+ * \f}
+ *
+ * In 1d we have:
+ *
+ * \f{align*}{
+ *
+ * J\frac{\partial \xi^{\hat{\imath}}}{\partial x^i}=1
```

Comment:
In 1d, should "i" -> "1"?

Also, I don't quite see how you get here from the above. Can you outline the intermediate steps (just as a github reply for now)?

Reply:
Sure. Consider a 1x1 matrix `A = [ a_{00} ]`. The determinant of this matrix is `a_{00}` and its inverse is `A^{-1} = [ 1 / a_{00} ]`. Thus, `det(A) A^{-1} = a_{00} / a_{00} = 1`. Make sense?

Reply:
Sorry, I wasn't clear.
- my first question was about the indices x^i, I think in 1d you can just write x^1
- my second question (I think this is what you are responding to) should have been phrased as: I don't understand how to turn the crank on the RHS of the general expression above and obtain the RHS here (= 1). I can also see the simple path you've laid out, but it won't generalize to the higher-dimensional cases.

Reply:
- Sorry, I agree with this and figured I'd just do it. Maybe do it in two steps? `(i expr) = (1 expr) = 1`?

- I don't think the general expression is valid in 1d because you can't define the Levi-Civita symbol in 1d.

Reply:
- 👍 
- Makes sense